From f9ea8f15c2754c50e12da0982d3cbe47f4f2d1e0 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 28 Aug 2014 02:06:27 -0400
Subject: [PATCH] ovsdb/automake.mk: mkdir ovsdb for target
 ovsdb/ovsdb-idlc.in

This fixes parallel build errors:

  /bin/bash: line 2: ovsdb/ovsdb-dot.tmp: No such file or directory
  make[2]: *** [ovsdb/ovsdb-idlc] Error 1

Target "ovsdb/ovsdb-idlc.in" needs to write "${B}/ovsdb/ovsdb-idlc.tmp",
but "${B}/ovsdb" is not there until below targets were called:

  * ovsdb/$(am__dirstamp), ovsdb/$(DEPDIR)/$(am__dirstamp)

These targets are generated by automake, we can't get "ovsdb/ovsdb-
idlc.in" depend on them directly. Just fix Makefile.am to create the
ovsdb/ before it was needed.

Reprocude the build errors:
  # update build/Makefile and run compile:
    ovsdb/$(am__dirstamp):
    +       sleep 30
            @$(MKDIR_P) ovsdb
            @: > ovsdb/$(am__dirstamp)
    ovsdb/$(DEPDIR)/$(am__dirstamp):
    +       sleep 30
            @$(MKDIR_P) ovsdb/$(DEPDIR)
            @: > ovsdb/$(DEPDIR)/$(am__dirstamp)

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 ovsdb/automake.mk |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ovsdb/automake.mk b/ovsdb/automake.mk
index 69e7900..ee89942 100644
--- a/ovsdb/automake.mk
+++ b/ovsdb/automake.mk
@@ -92,6 +92,12 @@ BUILT_SOURCES += $(OVSIDL_BUILT)
 # at least for now.
 $(OVSIDL_BUILT): ovsdb/ovsdb-idlc.in
 
+# Make sure the ovsdb/ is there while running target ovsdb/ovsdb-idlc.in
+mkdir_ovsdb:
+	@[ -d ovsdb ] || mkdir ovsdb
+
+ovsdb/ovsdb-idlc.in: mkdir_ovsdb
+
 # ovsdb-doc
 EXTRA_DIST += ovsdb/ovsdb-doc
 OVSDB_DOC = $(run_python) $(srcdir)/ovsdb/ovsdb-doc
-- 
1.7.9.5

